pinot-core/src/main/java/com/linkedin/pinot/core/crypt/PinotCrypterFactory.java

```diff
@@ -24,20 +28,54 @@
  */
 public class PinotCrypterFactory {
   public static final Logger LOGGER = LoggerFactory.getLogger(PinotCrypterFactory.class);
+  // Map of lower case simple class name to PinotCrypter object
+  private static Map<String, PinotCrypter> _crypterMap = new HashMap<>();
+  private static final String NOOP_PINOT_CRYPTER = "nooppinotcrypter";
+  private static final String CLASS = "class";
 
   // Prevent factory from being instantiated
   private PinotCrypterFactory() {
 
   }
 
+  /**
+   * Initializes map of crypter classes at startup time. Will initialize map with lower case simple class names.
+   * @param config
+   */
+  public static void init(Configuration config) {
+    // Get schemes and their respective classes
+    Iterator<String> keys = config.subset(CLASS).getKeys();
```

Comment:
Why does this method need to subset? Why doesn't the caller subset and pass the right config?